packages/slo-stat-bot/src/slo-lint.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GitHubAPI} from 'probot/lib/github';
+import Ajv, {ErrorObject} from 'ajv';
+import {Context} from 'probot';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const schema = require('./../data/schema.json');
+
+type Conclusion =
+  | 'success'
+  | 'failure'
+  | 'neutral'
+  | 'cancelled'
+  | 'timed_out'
+  | 'action_required'
+  | undefined;
+
+interface ValidationResults {
+  isValid: boolean;
+  errors?: ErrorObject[] | null;
+}
+
+interface PullsListFilesResponseItem {
+  filename: string;
+  sha: string;
+}
+
+/**
+ * Function gets list of files changed on the pr
+ * @param github unique installation id for each function
+ * @param owner of issue or pr
+ * @param repo of issue or pr
+ * @param pull_number of issue or pr
+ * @param per_page number of files that is listed per API call
+ * @returns an array of PullsListFilesResponseItem containing the filename and sha
+ */
+handle_lint.listFiles = async function listFiles(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  pull_number: number,
+  per_page: number
+): Promise<PullsListFilesResponseItem[] | null> {
+  try {
+    const listOfFiles = await github.pulls.listFiles({
+      owner,
+      repo,
+      pull_number,
+      per_page,
+    });
+    return listOfFiles.data;
+  } catch (err) {
+    console.warn(
```

Comment:
When do we use `console.error` vs `console.warn` vs `console.log`?  Maybe a better question for @bcoe :)